Core/clim-core/frames.lisp

```diff
@@ -936,10 +960,15 @@ documentation produced by presentations.")
                                 ,@(cdr top-level)))
           ,@geometry
           ,@user-default-initargs)
+         ,@(cond (metaclass
+                  `((:metaclass ,metaclass)))
+                 (update-instances-on-redefinition
+                  `((:metaclass redefinition-updates-instances-class))))
```

Comment:
wouldn't it be more straightforward to document the metaclass and provide an example for it (instead of adding a custom option to define-application-pane)?

Reply:
From my point of the view, the metaclass is an implementation detail. Ideally, we would figure out a way to allow a custom metaclass and `:update-instances-on-redefinition` at the same time.

Reply:
I will keep this discussion open to think about the "ideal" way you have mentioned when I dig deeper into the code.